File: Cf_Round_783/C_Make_it_Increasing.py
import math,sys;input=sys.stdin.readline;S=lambda:input().rstrip();I=lambda:int(S());M=lambda:map(int,S().split());L=lambda:list(M());mod1=1000000007;mod2=998244353
n=I();a=L();ans=math.inf
for i in range(n):
    c=0;prev=0
    for j in range(i-1,-1,-1):
        cur=((prev//a[j])+1)*a[j]
        c+=(prev//a[j])+1
        prev=cur
    prev=0
    for j in range(i+1,n):
        cur=((prev//a[j])+1)*a[j]
        c+=(prev//a[j])+1
        prev=cur
    ans=min(ans,c)
print(ans)

File: Cf_Round_783/C_Make_it_Increasing.java
import java.util.*;
public class C_Make_it_Increasing{
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        long n=sc.nextLong();
        long a[]=new long[(int) n];
        for(int i=0;i<n;i++) a[i]=sc.nextInt();
        long ans=10^18;
        for(int i=0;i<n;i++){
            long c=0, prev=0;
            for(int j=i-1;j>=0;j-=1){
                long cur=((prev/a[j])+1)*a[j];
                c+=(prev/a[j])+1;
                prev=cur;
            }
            prev=0;
            for(int j=i+1;j<n;j+=1){
                long cur=((prev/a[j])+1)*a[j];
                c+=(prev/a[j])+1;
                prev=cur;
            }
            ans=Math.min(ans,c);
        }
        sc.close();
        System.out.println(ans);
    }
}